src/services/ogc-features/src/main/java/com/camptocamp/opendata/ogc/features/autoconfigure/geotools/PostgisBackendAutoConfiguration.java

```diff
@@ -84,12 +93,47 @@ public void setDataSource(DataSource ds) {
 
         @Override
         protected @NonNull DataStore create() {
-            PostgisNGDataStoreFactory fac = new PostgisNGDataStoreFactory();
+            PostgisNGDataStoreFactory fac = new PostgisNGDataStoreFactory() {
+                @Override
+                protected SQLDialect createSQLDialect(JDBCDataStore dataStore) {
+                    return new SchemaUnawarePostGISDialect(dataStore);
+                }
+
+                @Override
+                protected SQLDialect createSQLDialect(JDBCDataStore dataStore, Map<String, ?> params) {
+                    return new SchemaUnawarePostGISDialect(dataStore);
+                }
+            };
             try {
                 return fac.createDataStore(connectionParams);
             } catch (IOException e) {
                 throw new UncheckedIOException(e);
             }
         }
     }
+
+    private static class SchemaUnawarePostGISDialect extends PostGISDialect {
+        public SchemaUnawarePostGISDialect(JDBCDataStore dataStore) {
+            super(dataStore);
+        }
+
+        @Override
+        public Integer getGeometrySRID(String schemaName, String tableName, String columnName, Connection cx)
+                throws SQLException {
+            Integer srid = null;
+            try (Statement statement = cx.createStatement()) {
+
+                String sqlStatement = "SELECT ST_SRID(\"" + columnName + "\")::int FROM \"" + tableName + "\" LIMIT 1";
```

Comment:
I'm wondering how this can be subject to SQL injections (disclaimer for other reviewers: I am the author of this code).

Reply:
I tried to tweak PreparedStmt without success. As columnName and table comes from the lib, I suppose that the breach isn't easily accessible.